lib/sensors/low_power_bms.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <io/i2c.hpp>
+
+static constexpr std::uint8_t kDefaultLpBmsAddress  = 0x10;
+
+namespace hyped::sensors {
+
+class LowPowerBMS {
+ public:
+  static std::optional<LowPowerBMS> create(core::ILogger &logger,
+                                           std::shared_ptr<io::II2c> i2c,
+                                           const std::uint8_t device_address);
+  ~LowPowerBMS();
+
+  std::optional<std::uint8_t> getStackVoltage();
+  std::optional<std::array<std::uint8_t, 16>> getCellData();
+
+ private:
+  LowPowerBMS(core::ILogger &logger,
+              std::shared_ptr<io::II2c> i2c,
+              const std::uint8_t device_address);
+  core::ILogger &logger_;
+  std::shared_ptr<io::II2c> i2c_;
+  const std::uint8_t device_address_;
+
+ private:
+  static constexpr std::uint8_t cell_voltages[] = {
```

Comment:
if this is the cell voltage register addresses then call it something like `kCellVoltageRegisters`